package.json

```diff
@@ -32,17 +32,17 @@
     "watch": "tsc -w"
   },
   "dependencies": {
-    "@jupyterlab/application": "^1.2.1",
-    "@jupyterlab/apputils": "^1.2.1",
-    "@jupyterlab/docregistry": "^1.2.1",
-    "@jupyterlab/notebook": "^1.2.2",
-    "@phosphor/disposable": "^1.3.1",
+    "@jupyterlab/application": "^2.0.2",
```

Comment:
In https://github.com/jupyterhub/jupyter-server-proxy/blob/80263c5748f1e0da19fe6b54e6b11883d141788f/jupyterlab-server-proxy/package.json#L32-L34 Erik went with `^1.0.0 || ^2.0.0` to allow the extension to work on 1.0 and 2.0. Can we do the same here?

Reply:
I don't think so. This repo includes UI components such as the `phosphor` to `lumino` toolkit switch which is a breaking change. The upgrade of font-awesome from 4 to 5 is also breaking as some icons were renamed, and I expect some changes to property names such as `iconClassName` to `iconClass` are breaking.

I read somewhere (can't find the link now) that if multiple versions of a JupyterLab extension exist it should automatically pick a compatible version, difficult to test though without actually releasing. Maybe @consideRatio knows?

Reply:
The question is if we should allow for this new version of the extension to be installed on JupyterLab 1. The benefit of doing so is that JupyterLab 1 can continue using the latest version of this extension, but the downside is that it may be incompatible with JupyterLab 1 or risk becoming incompatible with JupyterLab 1 because of tests only running on one version perhaps.

So, will this extension break on JupyterLab 1 if we make the changes in this PR? I think its fine with the rename of phosphor to lumino, but I there may have been another breaking change. I think the quickest path is to actually test it out, and if it is working still in JupyterLab 1, decide if it should be allowed to follow along for the ride or stay behind with less risk of breaking in the future.

Here is an example of me testing both versions of another extension: https://github.com/plotly/plotly.py/pull/2261#issuecomment-597087168

Here is an example of a PR I made to support both versions, along references to the migration guides in JupyterLabs documentation: https://github.com/plotly/plotly.py/pull/2245

Reply:
Jason Grout has helped out a lot in migration efforts I've been involved in so far, we could ask him for expert insight on this topic if we need it.

Reply:
To keep things simple (from a maintenance point of view) I'd be OK with people on lab 1.0 not being able to upgrade their version of this extension.

From a user's point of view it would be good to not allow them to install a version that won't work. Especially for users who type "install this thing without specifying a version"

Reply:
I tried setting the dependencies to
```js
  "dependencies": {
    "@jupyterlab/application": "^1.2",
    "@jupyterlab/apputils": "^1.2",
    "@jupyterlab/docregistry": "^1.2",
    "@jupyterlab/notebook": "^1.2",
    "@lumino/disposable": "^1.3.5",
    "@types/jquery": "^3",
    "jquery": "^3"
  },
```
to force the code changes in this PR to be built against JupyterLab 1.2. The build (`jlpm`) fails due to breaking changes in the UI components.

@consideratio do you know of a JupyterLab extension that was upgraded to JupyterLab 2 without backwards compatibility? We could verify that JupyterLab 1 will ignore the newer version of that extension.

Reply:
It is confirmed already by Jason Grout that JupyterLab will use package.json to find the latest candidate that have functional dependencies! You can release a new version with ^2 for the @jupyterlab/... dependencies and it will be ignored by JupyterLab 1 that will get stuck on old versions of the extension!

Reply:
@consideration one last (hopefully!) question. I've used `^2.0.2` instead of `^2` or `^2.0.0` because there were some minor bug fixes. Will this still behave as we're expecting: `>=2.0.2, <3`?

Reply:
Yepp this is all good I think! What matters is the content in package.json rather than package-lock.json when we run `jupyter labextension install <extension name>`. And the `^` sign means what you describe I believe, while `~2.0.2` would be `>=2.0.2, < 2.1.0` I think...

Oh wait, one difference perhaps regarding pre-releases, `^2` will exclude pre-releases, but hmmm I think the `>=2, <3` option would also.

Reply:
Thanks all! I think not supporting pre-releases is OK, it's complicated enough as it is 😀.

I'll merge this, add the tests https://github.com/manics/jupyter-offlinenotebook/pull/31, then release as `0.1.0`.